### PR TITLE
Safer handling of secrets data during configuration

### PR DESCRIPTION
**Pull request by nre-ableton**
I discovered a small problem with this wonderful Ansible role: writing Jenkins configuration data while the service is running is dangerous, but the service must be running to perform some of the setup tasks. This change splits the plugin configuration tasks from `configure-jenkins.yml` into a new file, and ensures that Jenkins is not running while the configuration data is changed. Also, I added a task which sets more restrictive permissions on the `secrets` folder when `jenkins_include_secrets` is used.

Specifically, Jenkins seems to keep some data in memory and flushes it to disk when it shuts down, which ends up overwriting the custom data copied there. The main culprit I found was `hudson.util.Secret`, which is one of the files that Jenkins uses to encrypt credential data. If this file is overwritten, any stored credentials in Jenkins global or job configuration files cannot be decrypted, and this causes various errors to occur when Jenkins attempts to use them.

On the subject of Jenkins credential storage, it's worth noting that the `secrets` directory isn't enough -- you also need the `secret.key` file to decrypt, which is stored in the top level of `JENKINS_HOME`. I considered adding a commit to this branch which would copy the file if `jenkins_include_secrets` was specified, but as `secret.key` is not stored in `secrets` I thought it would be too hacky to load this file from `{{ jenkins_source_secrets }}/..`. I guess it's an acceptable workaround to just copy that file in `jenkins_include_custom_files`, but it might be worth considering how this file can be taken care of if `jenkins_include_secrets` is given.

### CODE REVIEW

File: tasks/configure-jenkins.yml [hunk 1]
```diff
@@ -26,47 +36,8 @@
...
-  with_items: "{{ jenkins_custom_plugins }}"
-  when: jenkins_custom_plugins is defined
-
-- wait_for:
-    path: "{{ jenkins_home }}/plugins/{{ item }}.jpi"
-  with_items: "{{ jenkins_plugins }}"
+- name: set permissions on secrets dir
+  file:
+    path: "{{ jenkins_home }}/secrets"
+    mode: 0700
```
Comment (emmetog) [inline tasks/configure-jenkins.yml:42]: Let's explicitly set the owner and group of this directory to `{{ jenkins_config_owner }}` and `{{ jenkins_config_group }}`

Comment (emmetog) [reply tasks/configure-jenkins.yml:42]: In fact, can we just add `mode: 0700` to the `- name: secrets dir is up to date` task above?

**Comment by emmetog** [comment]
Thanks @nre-ableton for this PR!

We can remove the need for the task which sets the permissions on the secrets dir by adding the `mode: 0700` to the task above it, apart from this the PR looks great to me. If you can clean that up we'll get it merged

File: tasks/configure-jenkins.yml (hunk 1 above)
Comment (nre-ableton) [reply tasks/configure-jenkins.yml:42]: I tried doing that, but it set `0700` permissions on all the files *inside* `secrets` and not the directory itself. I'm a bit new to Ansible though, so there's a good chance that I could be missing something there. ;)

Comment (emmetog) [reply tasks/configure-jenkins.yml:42]: Ok, let's leave it like this then.

**Comment by emmetog** [comment]
Thanks again @nre-ableton!

**Comment by nre-ableton** [comment]
Thank you for this great project, it's been incredibly helpful!

**Pull request merged**